tests/test_spark.py

```diff
@@ -387,21 +387,23 @@ def test_driver_executor_tls():
 
 @pytest.mark.sanity
 def test_unique_vips():
+
+    @retrying.retry(stop_max_attempt_number=3, wait_fixed=2000)
+    def verify_ip_is_reachable(ip):
+        ok, _ = sdk_cmd.master_ssh("curl {}".format(ip))
+        assert ok
```

Comment:
what will happen if the assertion fails? is this type of error also handled by `retry`?

Reply:
Yup `retry` catches any exception that gets thrown and will retry based on params